### PR TITLE
For #5646  - Use new API for ETP Exceptions

### PR DESCRIPTION
**Pull request by ekager**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ekager** [comment]
We want to wait until https://github.com/mozilla-mobile/android-components/issues/4699 has a fix from GV before landing this

File: app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt [hunk 1]
```diff
@@ -10,17 +10,15 @@ import mozilla.components.browser.errorpages.ErrorType
...
-    override fun onLoadRequest(session: EngineSession, uri: String): RequestInterceptor.InterceptionResponse? {
-        val host = uri.tryGetHostFromUrl()
-
-        adjustTrackingProtection(host, context, session)
-
+    override fun onLoadRequest(
+        session: EngineSession,
+        uri: String
+    ): RequestInterceptor.InterceptionResponse? {
+        adjustTrackingProtection(context, session)
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt]: I'm looking for a way to get rid of this. At the moment we need it. I'm doing some local tests, I will let you know if I find an alternative.

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelDialogFragment.kt [hunk 2]
```diff
@@ -172,12 +170,24 @@ class TrackingProtectionPanelDialogFragment : AppCompatDialogFragment(), BackHan
...
-            }
-            with(context.components) {
-                useCases.sessionUseCases.reload.invoke(core.sessionManager.findSessionById(sessionId))
+            val useCase = TrackingProtectionUseCases(
+                sessionManager = context.components.core.sessionManager,
+                engine = context.components.core.engine
+            )
+            val session = context.components.core.sessionManager.findSessionById(sessionId)
+            session?.let {
+                useCase.containsException(it) { contains ->
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelDialogFragment.kt]: I'm not sure why we are checking if the session is on the list. I thought we can relay on `isEnabled` to determine which action to take. Is there any particular reason for checking?

File: app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt]: I think this code should live on the ac, I will file an issue for finding an alternative.

File: app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelDialogFragment.kt (hunk 2 above)
Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelDialogFragment.kt]: Nope you're right :)

**Comment by Amejia481** [comment]
Apart of the nits: that I mentioned above, it looks pretty good  👍 .
At the moment, we don't have a good solution for the interceptor issue, we have to keep that, until we find a better solution on ac.

**Comment by ekager** [comment]
We can land this PR now since in #6354 we've decided on hiding just the singular deletion from the exceptions list

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt [hunk 3]
```diff
@@ -151,10 +152,25 @@ class QuickSettingsSheetDialogFragment : AppCompatDialogFragment() {
...
+    private fun toggleTrackingProtection(context: Context) {
+        val useCase = TrackingProtectionUseCases(
+            sessionManager = context.components.core.sessionManager,
+            engine = context.components.core.engine
+        )
+        val session = context.components.core.sessionManager.findSessionById(sessionId)
+        session?.let {
+            useCase.containsException(it) { contains ->
+                if (contains) {
+                    useCase.removeException(it)
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt]: Should we  also track when the user remove an exception?

Comment (ekager) [reply app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt]: We would need a new telemetry probe and data review

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsSheetDialogFragment.kt]: I don't really know if we need  it just want to confirm :)

File: app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/AppRequestInterceptor.kt]: This is the ac issue https://github.com/mozilla-mobile/android-components/issues/4935
We are going to work on this issue next sprint.

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 4]
```diff
@@ -224,14 +225,21 @@ class BrowserFragment : BaseBrowserFragment(), BackHandler {
...
-        val directions =
-            BrowserFragmentDirections.actionBrowserFragmentToTrackingProtectionPanelDialogFragment(
-                sessionId = session.id,
-                url = session.url,
-                trackingProtectionEnabled = session.trackerBlockingEnabled,
-                gravity = getAppropriateLayoutGravity()
-            )
-        nav(R.id.browserFragment, directions)
+        val useCase = TrackingProtectionUseCases(
+            sessionManager = context!!.components.core.sessionManager,
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: nit: We could use  `requireContext()` instead of ‼️ :)

**Pull request merged**